### PR TITLE
Closes AgileVentures/MetPlus_tracker#67

### PR DESCRIPTION
**Pull request by vijayakarumudi**
PR for AgileVentures/MetPlus_PETS/#67

### CODE REVIEW

File: app/models/job_category.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class JobCategory < ActiveRecord::Base
+ has_and_belongs_to_many :agency_people, join_table: :job_specialities
+ has_many :jobs
+ has_and_belongs_to_many:skills
```
Comment (joaopapereira) [inline app/models/job_category.rb]: Missing space between words

Comment (joaopapereira) [inline app/models/job_category.rb]: Aren't we missing the relational table job_category_skills here?

File: db/migrate/20151027162612_create_job_categories.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CreateJobCategories < ActiveRecord::Migration
+  def change
+    create_table :job_categories do |t|
+      t.string :name
+      t.string :description
+      t.belongs_to :jobs, index: true, foreign_key: true
+      t.belongs_to :skills, index: true, foreign_key: true
```
Comment (joaopapereira) [inline db/migrate/20151027162612_create_job_categories.rb]: This is incorrect, we should have a table to create this relationship

File: db/migrate/20151027162612_create_job_categories.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class CreateJobCategories < ActiveRecord::Migration
+  def change
+    create_table :job_categories do |t|
+      t.string :name
+      t.string :description
+      t.belongs_to :jobs, index: true, foreign_key: true
```
Comment (joaopapereira) [inline db/migrate/20151027162612_create_job_categories.rb]: Shouldn't this be in the Jobs table?

File: app/models/job_category.rb (hunk 1 above)
Comment (patmbolger) [inline app/models/job_category.rb]: We're ok.  That table is a simple join table, used by the *has_and_belongs_to_many: skills* association.

File: spec/models/job_category_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe JobCategory, type: :model do
+
+   
+   describe 'Associations' do
+    
+     it { is_expected.to have_and_belong_to_many :skills }
+     it { is_expected.to have_many :jobs }
+     xit{ is_expected.to have_many_and_belong_to(:agency_people).join_table(:job_specialities)}
```
Comment (joaopapereira) [inline spec/models/job_category_spec.rb]: This association table is already present and is called job_specialities  sor we can uncomment this test

**Comment by joaopapereira** [comment]
Please review my comments

File: db/migrate/20151027162612_create_job_categories.rb (hunk 2 above)
Comment (patmbolger) [inline db/migrate/20151027162612_create_job_categories.rb]: The association to 'skills' goes through a join table - we don't want a foreign key here for skills.

Comment (patmbolger) [inline db/migrate/20151027162612_create_job_categories.rb]: Also, JobCategory has a *has_many* association with Job - no foreign key needed here for jobs.

File: spec/models/job_category_spec.rb (hunk 4 above)
Comment (vijayakarumudi) [inline spec/models/job_category_spec.rb]: I finished all the comments related tasks

**Pull request closed without merging**